guides-and-standards/guide-choosing-software-dependencies.md

```diff
@@ -0,0 +1,59 @@
+- Tools designed to analyze experimental data in high throughput like [arcadia-phenotypeomat-protocol]([https://github.com/Arcadia-Science/arcadia-phenotypeomat-protocol)
+
+## Things to look for when deciding to include a tool as a dependency
+
+A dependency is any software tool that is required to make your tool work. For example, BLAST is a dependency of ProteinCartography, and Nextflow and OrthoFinder are dependencies of NovelTree. Below we include some signals for assessing whether a software tool is high quality.
+
+### Critical — we can’t use the tool if these criteria are not met
+
+- **The tool has a license and the license is permissible for commercial use.** Common licenses like BSD2, BSD3, GPL3, and MIT can all be used. Some other licenses can be used as well.
+  - Hint: you can search for the word "commercial" in the license. This will often take you to text about whether the software can be used commercially. If it can’t, it will usually say something like "for non-commercial use only."
+- **The tool can be run on the command line without human input.** A Graphical User Interface (GUI) (including web applications like Streamlit) is not required to run the tool. The tool isn’t encoded in notebooks.
+- **The tool can be run on a Linux platform.** This is a requirement\* for things dockerization and for running on AWS.
+  - *It’s not actually a*requirement\* — the Mac operating systems can’t really be dockerized, but Windows can. Similarly, there are Windows operating systems on AWS but these are sometimes annoying to use.
+
+### Important — think carefully before proceeding with the tool if these criteria are not met
+
+- **The software code is on GitHub or another popular developer platform like GitLab or BitBucket.** Using these platforms signals that the tool is open source and likely versioned. Sometimes you'll find code on SourceForge. This is ok and potentially workable, but not the best. Sometimes you'll find code in the supplement of a paper. This is not awesome and probably signals that the tool will be hard to install or use but you can still try if you really want to.
+- **The tool is easily installable.** To be able to use a tool, we need to be able to install it. Below I’ve included somewhat arbitrary gradations of what makes something easy to install. You can check these things by eye to quickly see if you think a tool will be installable, but often the best way to determine this is to install and use the tool.
+  - Very easy: cross-platform installation as a conda package
+  - Easy: installable from another package repository with another package manager like pip (python) or `install_packages()` (R)
+  - Somewhat easy: Installable from GitHub with a package manager
+  - Somewhat easy: Has a dockerfile and an available Docker container
+  - Somewhat difficult: has an environment yaml file or requirements.txt file that lists versioned dependencies and has installation instructions that involve cloning the repository
```

Comment:
My personal opinion, but I think for most Arcadians cloning a repository and creating an environment based on a yaml/requirements.txt file would be easier than dockerfile/Docker container. Trying to help a couple people install and run Docker on their machines I've learned is difficult.

Reply:
hmm yes i agree, but I'm thinking of ease of porting the installation over to a high throughput workflow like thing such as nextflow. If someone likes a tool that is run in docker and can test it, its easier to port this over to something that can run on AWS batch eventually. Given that, should i amend this?

Reply:
Oh I see, yeah then I would keep that the same. Again I think my brain was shifting while reading this between if it can it just be used reproducibly vs in a larger high-throughput tool. Maybe some justification at the beginning of this could be added to clarify that? And in theory if somebody were to go through this guide, who is the audience/decision maker they are keeping in mind? Are the people mentioned, for example, vetting out tools so they themselves can build a workflow or suggest to somebody like you or myself to build a workflow implementing that tool?

Reply:
kk i'll add some justification in the beginning to make this more clear.

Reply:
added "In these sorts of projects, we expect that the tools we use may one day be incorporated into a workflow or package" as the second sentence!